### DISCUSSION TITLE
Question regarding dependency error messages

### DISCUSSION THREAD

**Discussion by wdfinch** [yarnpkg/berry#2611] [Q&A] [answered]
Hi. I am trying to compile some code in a workspace via typescript and I'm running into these minor errors. It seems that typescript cannot find the module 'prop-types' even though this is listed as a dependency by packages in the workspaces that need it and it is present inside my .caches folder.

I'm wondering if anyone could explain what is happening here and what I can do to resolve these messages. This seems like a yarn 2 specific issue as yarn 1 did not run into these errors. 

.yarn/$$virtual/react-svg-virtual-cef9eff21d/0/cache/react-svg-npm-12.1.0-ee18b746f3-31636e51bc.zip/node_modules/react-svg/dist/ReactSVG.d.ts(1,28): error TS7016: Could not find a declaration file for module 'prop-types'. '/home/wfinch/work/dialsource1/dialsource/frontend/.yarn/cache/prop-types-npm-15.7.2-d7a04f2274-a440dd406c.zip/node_modules/prop-types/index.js' implicitly has an 'any' type.

 If the 'prop-types' package actually exposes this module, consider sending a pull request to amend 'https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/prop-types`
.yarn/cache/react-ripples-npm-2.2.1-a959bfa856-eca16e1bb9.zip/node_modules/react-ripples/cjs/createRipples.d.ts(2,23): error TS2307: Cannot find module 'prop-types' or its corresponding type declarations.

**Comment by merceyz** [comment, marked as answer]
It's complaining about `Could not find a declaration file for module 'prop-types'` so that means it's after the types for the package, since `prop-types` doesn't provide it's own that would be `@types/prop-types`, so whatever depends on `prop-types` lacks a dependency on `@types/prop-types` as well. You can use [`packageExtensions`](https://yarnpkg.com/getting-started/migration/#fix-dependencies-with-packageextensions) to fix that.

> This seems like a yarn 2 specific issue as yarn 1 did not run into these errors.

`node_modules`* - Hoisting hides all kinds of undeclared dependency issues

**Reply by wdfinch** [reply]
Thanks for the reply! I posted this when I was still new to yarn 2 so I was able to figure this out. I appreciate you getting back and this will for sure help new users who are confused by this error.
